# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2ContextQueries.cpp
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGL2ContextQueries.cpp
--- a/dom/canvas/WebGL2ContextQueries.cpp	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGL2ContextQueries.cpp	Wed Jan 08 22:19:23 2020 +0000
@@ -91,7 +91,7 @@
   const auto& slot = ValidateQuerySlotByTarget(target);
   if (!slot) return;
 
-  const auto& query = *slot;
+  const auto query = *slot;  // Grab a strong reference.
   if (!query) return ErrorInvalidOperation("Query target not active.");
 
   query->EndQuery();